applications/asset_tracker_v2/tests/cloud_codec_nrf/src/nrf_cloud_codec_test.c

```diff
@@ -0,0 +1,686 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <unity.h>
+#include <stdbool.h>
+
+#include "cloud_codec.h"
+#include <cJSON_os.h>
+#include <errno.h>
+
+struct cloud_codec_data codec;
+int ret;
+
+void setUp(void)
+{
+	ret = cloud_codec_init(NULL, NULL);
+	TEST_ASSERT_EQUAL(0, ret);
+
+	memset(&codec, 0, sizeof(codec));
+}
+
+void tearDown(void)
+{
+	cJSON_FreeString(codec.buf);
+}
+
+/* Suite teardown shall finalize with mandatory call to generic_suiteTearDown. */
+extern int generic_suiteTearDown(int num_failures);
+
+int test_suiteTearDown(int num_failures)
+{
+	return generic_suiteTearDown(num_failures);
+}
+
+#define CONF_RECV_EXAMPLE \
+"{"\
+	"\"config\":{"\
+		"\"activeMode\":false,"\
+		"\"gnssTimeout\":60,"\
+		"\"activeWaitTime\":120,"\
+		"\"movementResolution\":120,"\
+		"\"movementTimeout\":3600,"\
+		"\"accThreshAct\":10,"\
+		"\"accThreshInact\":5,"\
+		"\"accTimeoutInact\":1,"\
+		"\"nod\":["\
+			"\"gnss\","\
+			"\"ncell\""\
+		"]"\
+	"}"\
+"}"
+
+#define CONF_SEND_EXAMPLE "{\"state\":{\"reported\":" CONF_RECV_EXAMPLE "}}"
+
+#define UI_EXAMPLE \
+"{"\
+	"\"data\":\"1\","\
+	"\"appId\":\"BUTTON\","\
+	"\"messageType\":\"DATA\","\
+	"\"ts\":1563968747123"\
+"}"
+
+#define BAT_BATCH_EXAMPLE \
+"[{"\
+	"\"appId\":\"VOLTAGE\","\
+	"\"messageType\":\"DATA\","\
+	"\"ts\":1563968747123,"\
+	"\"data\":\"50\""\
+"}]"
+
+/** TODO: this differs from the schemas
+ * https://github.com/nRFCloud/application-protocols/blob/v1/schemas/deviceToCloud/gnss/GPGGA-example.json
+ */
+#define GNSS_BATCH_EXAMPLE \
+"[{"\
+	"\"appId\":\"GNSS\","\
+	"\"messageType\":\"DATA\","\
+	"\"ts\":1563968747123,"\
+	"\"data\":\"$GPGGA,141856.00,6326.27336,N,01028.08721,E,1,09,0.81,34.7,M,39.8,M,,*64\\n\""\
+"}]"
+
+/** TODO: this differs from the schema
+ * https://github.com/nRFCloud/application-protocols/blob/v1/schemas/deviceToCloud/device/device.json
+ */
+#define MODEM_STATIC_BATCH_EXAMPLE \
+"[{"\
+	"\"appId\":\"DEVICE\","\
+	"\"messageType\":\"DATA\","\
+	"\"ts\":1563968747123,"\
+	"\"data\":{"\
+		"\"deviceInfo\":{"\
+			"\"imei\":\"352656106111232\","\
+			"\"iccid\":\"89450421180216211234\","\
+			"\"modemFirmware\":\"mfw_nrf9160_1.2.3\","\
+			"\"board\":\"nrf9160dk_nrf9160\","\
+			"\"appVersion\":\"v1.0.0-development\""\
+		"}"\
+	"}"\
+"}]"
+
+/** TODO: this differs from the schema
```

Comment:
Are the TODOs fixed?

Reply:
there are two superfluous elements (gnss ts and rsrp), but I think we can ignore them

Reply:
I removed those todo comments